src/client.ts

```diff
@@ -296,10 +296,11 @@ export class Client {
   }
 
   #getSecret(partialClientConfig?: ClientConfiguration): string {
-    let fallback = undefined;
-    if (typeof process === "object") {
-      fallback = process.env["FAUNA_SECRET"];
-    }
+    const fallback =
+      process?.env !== null && typeof process.env === "object"
+        ? process.env["FAUNA_SECRET"]
+        : undefined;
```

Comment:
can we just do this?

```suggestion
    const fallback = process?.env?.FAUNA_SECRET;
```